src/test/bridges/notional/NotionalE2E.t.sol

```diff
@@ -0,0 +1,85 @@
+        IWrappedfCashFactory(0x5D051DeB5db151C2172dCdCCD42e6A2953E27261);
+    NotionalViews public constant NOTIONAL_VIEW = NotionalViews(0x1344A36A1B56144C3Bc62E7757377D288fDE0369);
+    // To store the id of the stability pool bridge after being added
+    uint256 private id;
+
+    function setUp() public {
+        // Deploy a new stability pool bridge and set approvals
+        bridge = new NotionalBridgeContract(address(ROLLUP_PROCESSOR));
+
+        // Use the label cheat-code to mark the address in the traces
+        vm.label(address(bridge), "NotionalBridge");
+
+        // Impersonate the multi-sig to add a new bridge
+        vm.startPrank(MULTI_SIG);
+
+        // List the staking-bridge with a gasLimit of 350k
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 1000000);
+
+        vm.stopPrank();
+
+        // Fetch the id of the notional bridge
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+    }
+
+    function testFullDepositWithdrawalE2E(uint256 _depositAmount) public {
+        _depositAmount = bound(_depositAmount, 1e16, 1e21);
+        uint40 maturity = uint40(NOTIONAL_VIEW.getActiveMarkets(ETH_CURRENCY_ID)[0].maturity);
+        address fcash = FCASH_FACTORY.deployWrapper(ETH_CURRENCY_ID, maturity);
+        // Use the helper function to fetch Aztec assets
+        // Add support for our fcash token
+        vm.prank(MULTI_SIG);
+        ROLLUP_PROCESSOR.setSupportedAsset(fcash, 200000);
+
+        AztecTypes.AztecAsset memory ethAsset = getRealAztecAsset(ETH);
+        AztecTypes.AztecAsset memory fcashAsset = getRealAztecAsset(fcash);
+
+        address[] memory tokens = new address[](1);
+        tokens[0] = fcash;
+        bridge.approve(tokens, address(ROLLUP_PROCESSOR));
+        tokens[0] = WETH;
+        bridge.approve(tokens, fcash);
+
+        // DEPOSIT
+        vm.deal(address(ROLLUP_PROCESSOR), _depositAmount);
+        // Compute deposit calldata
+        uint256 bridgeCallData = encodeBridgeCallData(id, ethAsset, emptyAsset, fcashAsset, emptyAsset, 1);
+
+        (uint256 outputValueA, , ) = sendDefiRollup(bridgeCallData, _depositAmount);
+        assertEq(
+            ERC20(fcash).balanceOf(address(ROLLUP_PROCESSOR)),
+            outputValueA,
+            "Incorrect fcash balance of rollup processor"
+        );
+        // WITHDRAWAL
+        // Compute withdrawal calldata
+        bridgeCallData = encodeBridgeCallData(id, fcashAsset, emptyAsset, ethAsset, emptyAsset, 0);
+
+        uint256 ethBalanceBefore = address(ROLLUP_PROCESSOR).balance;
+
+        (outputValueA, , ) = sendDefiRollup(bridgeCallData, ERC20(fcash).balanceOf(address(ROLLUP_PROCESSOR)));
+        uint256 ethBalanceAfter = address(ROLLUP_PROCESSOR).balance;
+        uint256 ethWithdrew = ethBalanceAfter - ethBalanceBefore;
+        assertGt(ethWithdrew, 0, "Zero withdraw amount");
+        assertGt(ethWithdrew * 10000, _depositAmount * 9900, "Should take most of money back");
+    }
```

Comment:
Please also add E2E tests of ERC20 tokens. This is necessary so that we properly test approvals and gas limits.